### PR TITLE
Support background color on MacOS

### PR DESCRIPTION
**Pull request by Matts966**
## What kind of change does this PR introduce?
- Feature
- Documentation

## Did this PR introduce a breaking change? 
_A breaking change includes anything that breaks backwards compatibility either at compile or run time._
- No

Support colorized title bar by `g:neovide_background_color` for macOS, discussed in #1298 

BackgroundColor

### CODE REVIEW

**Comment by Matts966** [comment]
https://user-images.githubusercontent.com/28551465/183221643-2a74287b-9c1f-4a54-bd3b-8233c9452daf.mov

File: src/window/mod.rs [hunk 1]
```diff
@@ -18,6 +18,7 @@ use tokio::sync::mpsc::UnboundedReceiver;
 
 #[cfg(target_os = "macos")]
 use glutin::platform::macos::WindowBuilderExtMacOS;
+use crate::settings::draw_background;
```
Comment (MultisampledNight) [inline src/window/mod.rs:21]: If you intended to have this import only active if the target OS is `macos`, note that `#[...]` attributes only apply to the next item.

File: website/docs/configuration.md [hunk 2]
```diff
@@ -38,6 +38,25 @@ let g:neovide_transparency=0.8
...
+let g:neovide_background_color = '#0f1117'.printf('%x', float2nr(255 * g:neovide_transparency_point))
+
+" Add keybinds to change transparency
+function! ChangeTransparency(delta)
+  let g:neovide_transparency_point = g:neovide_transparency_point + a:delta
+  let g:neovide_background_color = '#0f1117'.printf('%x', float2nr(255 * g:neovide_transparency_point))
+endfunction
+noremap <expr><D-]> ChangeTransparency(0.01)
+noremap <expr><D-[> ChangeTransparency(-0.01)
+```
```
Comment (MultisampledNight) [inline website/docs/configuration.md:56]: While it's sure nice to have a whole ready-to-use machine for changing the transparency, this page is more or less only intended to show the name of those config values, what they do and their defaults. The mechanism allowing to change this through a keybinding might have a better home in `website/docs/faq.md`- while we should probably call that thing more like a "tips & tricks" section instead.

File: src/settings/draw_background.rs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+use cocoa::appkit::{NSWindow, NSColor};
+use cocoa::base::{id, nil};
+use objc::{
+    rc::autoreleasepool,
+    runtime::{YES},
+};
+use csscolorparser::Color;
+
+#[cfg(target_os = "macos")]
+pub fn draw_background(window: &WindowedContext<PossiblyCurrent>) {
```
Comment (MultisampledNight) [inline src/settings/draw_background.rs:15]: For compartization's sake, I'm not sure if this window logic belongs into `src/settings`. It feels more like it should go into `src/window` instead.

**Review by MultisampledNight — COMMENTED** [review]
Thank you! Three small things I'd like to have addressed before driving this thing home.

File: src/window/mod.rs (hunk 1 above)
Comment (Matts966) [reply src/window/mod.rs]: I used code below to import the function only on macOS.
Thank you!

```rust
#[cfg(target_os = "macos")]
mod draw_background;
```

File: website/docs/configuration.md (hunk 2 above)
Comment (Matts966) [reply website/docs/configuration.md]: I separated them. Thank you!

File: src/settings/draw_background.rs (hunk 3 above)
Comment (Matts966) [reply src/settings/draw_background.rs]: I moved it. Thank you !

**Comment by Matts966** [comment]
@MultisampledNight 
Thank you for your nice review :)
I fixed your points and confirmed `cargo fmt --check` exit with code 0.

**Comment by MultisampledNight** [comment]
Alright then, I think this should be ready to go. Thank you again!

One small thing for the future in case you contribute again somewhere (also applies not only to this repo): Consider making a branch on your fork for every PR you make, and don't directly commit to your fork's main. This avoids some tricky local git scenarios where technically both origin and your remote are checked out, but both branches are named `main`.

**Comment by Matts966** [comment]
@MultisampledNight 
Thank you for your kind fix 🥹

**Comment by zhaozg** [comment]
I got error on MacOS X86_64

```
[I] ✘101 ➜ PYTHON=python3 cargo build --release
   Compiling neovide v0.9.0 (/Users/zhaozg/work/extra/vide)
error[E0433]: failed to resolve: use of undeclared crate or module `cocoa`
 --> src/window/draw_background.rs:4:5
  |
4 | use cocoa::appkit::{NSColor, NSWindow};
  |     ^^^^^ use of undeclared crate or module `cocoa`

error[E0433]: failed to resolve: use of undeclared crate or module `cocoa`
 --> src/window/draw_background.rs:5:5
  |
5 | use cocoa::base::{id, nil};
  |     ^^^^^ use of undeclared crate or module `cocoa`

error[E0433]: failed to resolve: use of undeclared crate or module `objc`
 --> src/window/draw_background.rs:8:5
  |
8 | use objc::{rc::autoreleasepool, runtime::YES};
  |     ^^^^ use of undeclared crate or module `objc`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `neovide` due to 3 previous errors

```

**Comment by MultisampledNight** [comment]
Oh welp, I'll try to take care of it.

**Comment by MultisampledNight** [comment]
I think I got it. Could you try again?

**Comment by zhaozg** [comment]
> I think I got it. Could you try again?

It's works, thank you.

**Pull request merged**